snuba/querylog/factory.py

```diff
@@ -0,0 +1,34 @@
+from snuba.clickhouse.query_dsl.accessors import get_time_range
```

Comment:
I don't think we need these factory files. I think these functions can live in the validation file.

Reply:
My thinking was that the create_snuba_metadata_query function is related to constructing SnubaMetadataQuery objects, rather than doing validation, which is why I thought it made more sense to be located with the SnubaMetadataQuery files.

One other thought I had is that the same construction logic is repeated here https://github.com/getsentry/snuba/blob/master/snuba/web/query.py#L102
this is where I took the logic from, I didnt change it to use the factory new factory function, but I probably should in order to avoid repeated code.

If I keep create_snuba_metadata_query in factory.py or perhaps query_metadata.py it would be easy to use in both places. If I make it a private method in validation.py, I couldnt do this.

Reply:
I made the change to remove the duplicate code, so you can see what i was talking about